tools/storage/src/restore.rs

```diff
@@ -0,0 +1,31 @@
+// Do a full restoration given a RestoreBundle with verified manifests
+
+use std::path::Path;
+
+use crate::{
+    dbtool_init::{run_restore, RestoreTypes},
+    restore_bundle::RestoreBundle,
+};
+
+pub async fn full_restore(target_db: &Path, bundle: &RestoreBundle) -> anyhow::Result<()> {
+    assert!(
+        bundle.is_loaded(),
+        "the restore bundle hasn't been checked yet"
+    );
+
+    run_restore(RestoreTypes::Epoch, target_db.to_owned(), bundle).await?;
+    run_restore(RestoreTypes::Snapshot, target_db.to_owned(), bundle).await?;
+    run_restore(RestoreTypes::Transaction, target_db.to_owned(), bundle).await?;
+    Ok(())
+}
+
+#[tokio::test]
```

Comment:
If possible please add here too an assert statement to verify that the expected outcome matches the actual result

Reply:
Same as above. Done. Checking that the expected DB directories were created.